packages/datadog-plugin-mongodb-core/src/index.js

```diff
@@ -55,8 +55,8 @@ class MongodbCorePlugin extends Plugin {
 
 function getQuery (cmd) {
   if (!cmd || typeof cmd !== 'object' || Array.isArray(cmd)) return
-  if (cmd.query) return JSON.stringify(sanitize(cmd.query))
-  if (cmd.filter) return JSON.stringify(sanitize(cmd.filter))
+  if (cmd.query) return JSON.stringify(cmd.query)
```

Comment:
How does this handle circular dependencies or extremely large queries?

Reply:
Don't know about large queries, but pretty sure circular references don't make sense in queries and mongodb-core itself would just crash. 🤷🏻 

How do we want to handle this? Maybe keep the sanitize method and only do the replacement when the depth limit is reached and maybe truncate the final string?

Reply:
Restructured it a bit. Does this fit what you have in mind?

Reply:
> Don't know about large queries, but pretty sure circular references don't make sense in queries and mongodb-core itself would just crash.

It does make sense and it does happen unfortunately. The depth limiting logic was added because of that very issue.

> How do we want to handle this? Maybe keep the sanitize method and only do the replacement when the depth limit is reached and maybe truncate the final string?

I'm not actually sure, but I'd rather keep the logic as simple as possible, so maybe just handling depth and special values.

> Restructured it a bit. Does this fit what you have in mind?

Added comments.